homeassistant/components/balboa/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for Balboa Spa Client integration."""
+from pybalboa import BalboaSpaWifi
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.core import callback
+
+from .const import _LOGGER, CONF_SYNC_TIME, DOMAIN
+
+DATA_SCHEMA = vol.Schema(
+    {vol.Required(CONF_HOST): str, vol.Required(CONF_NAME, default="Spa"): str}
+)
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect."""
+    for entry in hass.config_entries.async_entries(DOMAIN):
+        if entry.data[CONF_HOST] == data[CONF_HOST]:
+            raise AlreadyConfigured
+
+    _LOGGER.debug("Attempting to connect to %s", data[CONF_HOST])
+    spa = BalboaSpaWifi(data[CONF_HOST])
+    connected = await spa.connect()
+    _LOGGER.debug("Got connected = %d", connected)
+    if not connected:
+        raise CannotConnect
+    await spa.disconnect()
+
+    return {"title": data[CONF_NAME]}
+
+
+class BalboaSpaClientFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a Balboa Spa Client config flow."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_PUSH
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(config_entry):
+        """Get the options flow for this handler."""
+        return BalboaSpaClientOptionsFlowHandler(config_entry)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+        errors = {}
+        if user_input is not None:
+            try:
+                info = await validate_input(self.hass, user_input)
```

Comment:
Can we get the `macaddr` and use it as the config entry unique id?

https://github.com/garbled1/pybalboa/blob/master/pybalboa/balboa.py#L164

Then we can do
`await self.async_set_unique_id(formatted_mac)`
`self._abort_if_unique_id_configured()`

Reply:
https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#unique-id-requirements

Reply:
I don't have an easy way to get the macaddr there, unless there is some kind of reverse lookup arp facility in HA.  Is there?

Reply:
It looks like the lib will populate the macaddr with this call.  Since you are already connected here it seems like that should be a good way to get it and validate the connection is working.

https://github.com/garbled1/pybalboa/blob/master/pybalboa/balboa.py#L254
https://github.com/garbled1/pybalboa/blob/master/pybalboa/balboa.py#L855